src/pages/CommitDetailPage/CommitCoverage/routes/ComponentsSelector/ComponentsSelector.tsx

```diff
@@ -19,39 +19,41 @@ function ComponentsSelector() {
     params?.components
   )
   const { data, isLoading } = useCommitComponents({
-    filters: {
-      components: componentSearch ? [componentSearch] : undefined,
-    },
     options: {
       suspense: false,
     },
   })
 
-  const components = data?.components
+  const components = useMemo(() => {
+    return data?.components
+  }, [data])
+
+  const componentNames = useMemo(() => {
```

Comment:
Just curious if you already tried out the regExp class and then regExp.test('',param2) against it. I haven't on the Frontend personally but maybe it'd be another slightly more robust option that isn't too much of a pain to implement

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test